### PR TITLE
Observe specific enables/disables even if followed by "all"

### LINKED ISSUE #3696: enable/disable options in rcfile should not depend on the order with which they are specified
Hello,
i'm running `pylint 2.5.3 / astroid 2.4.2`; i kinda understand why

`$ pylint --enable=all --disable=fixme`

behaves differently than

`$ pylint --disable=fixme --enable=all`

(where the first command enables everything and then disable `fixme`, while the second command the disable option is overwritten by `enable=all`) but i dont think it should be the same in the rcfile: the `disable` section is (by default) before the `enable` section, so if i want to have the same effect of command1 i need to swap the sections around.

on the cli i can specify multiple enable/disable options, but that's not allowed in the rcfile, so the 
 current result is extremely counter-intuitive; and rcfile with

```
disable=fixme
enable=all
```

is clear what result the user wants: to enable all checks _except_ for some, and i shouldnt need to move config options around.

can you please get that fixed?

Thanks!

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |

## Description
Let individual enables/disables take effect even if followed by `--enable=all` or `--disable=all`. Previously, this was order-specific, which is tolerable on the CLI but inconvenient for configuration files.

This means that for:
```
pylint my_module --enable=fixme --disable=all
```
...you'll now get the fixme messages, whereas before, you'd have to reverse the order of the arguments.  (If the new result is not what you want, you can get the prior behavior by removing the superfluous first argument.)

Order-dependence still applies for message categories such as "R", "W", etc. This PR just moves the `=all` arguments to the front.

Providing both `--enable=all` and `--disable=all` in any order now raises an exception.

Closes #3696

### CODE REVIEW

**Review by Pierre-Sassoulas — DISMISSED** [review]
I think it make sense, great work ! We might want to extract the logic tp be able to use all in inline comment later ?

File: pylint/config/config_initialization.py [hunk 1]
```diff
@@ -147,3 +154,48 @@ def _config_initialization(
             for arg in parsed_args_list
         )
     )
+
+
+def _order_all_first(config_args: list[str], *, joined: bool) -> list[str]:
```
Comment (DanielNoord) [inline pylint/config/config_initialization.py:159]: Did you consider making this part of `ArgumentsManager`? It feels like we want to make this the default behaviour and don't want it to be possible to parse arguments without this behaviour right?

Comment (DanielNoord) [reply pylint/config/config_initialization.py:159]: We do so here:
https://github.com/pylint-dev/pylint/blob/f815e9f5fcdd5caffc04e335fc4590a65989ef21/pylint/pyreverse/main.py#L279
Although `all` isn't used here.

I also think we might skip in some test setups.

I'm not sure, we do quite a bit of parsing in `config_initialization` as well and perhaps `ArgumentsManager` should be kept small. Let's see what Pierre thinks.

File: tests/config/test_functional_config_loading.py [hunk 2]
```diff
@@ -43,6 +43,11 @@
     str(path.relative_to(FUNCTIONAL_DIR))
     for ext in ACCEPTED_CONFIGURATION_EXTENSIONS
     for path in FUNCTIONAL_DIR.rglob(f"*.{ext}")
+    if (str_path := str(path))
+    # The enable/disable all tests are not practical with this framework.
+    # They require manually listing ~400 messages, which will
```
Comment (Pierre-Sassoulas) [inline tests/config/test_functional_config_loading.py:48]: Hurgh, yeah, too bad.

File: pylint/config/config_initialization.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply pylint/config/config_initialization.py:159]: I'm also not very clear on the design if the argument parser, I usually have simple argparse add_argument calls with custom type if required in my code. The current design might be due to optparse being used before or the need to pre-parse some if the arguments ? Anyway, it seems to work as is, so I don't think committing too much in the current design or a refactor of the current design or this merge request would be worth it.

**Review by Pierre-Sassoulas — APPROVED** [review]
LGreatTM :)

**Pull request merged**